http.h

```diff
@@ -49,22 +51,24 @@ struct http_get {
 	int code;
 };
 
-int http_parse_uri(const char *uri, struct http_uri *u);
+int parse_uri(const char *uri, struct http_uri *u);
 
 /* frees contents of @u, not @u itself */
 void http_free_uri(struct http_uri *u);
 
-int http_open(struct http_get *hg, int timeout_ms);
+int socket_open(struct http_get *hg, int timeout_ms);
```

Comment:
I would hide everything socket away to only use connection. Make it private.

Only let fd be part of connection.